src/components/Search/SearchPageHeader/SearchPageHeaderInput.tsx

```diff
@@ -24,91 +29,57 @@ import {getAutocompleteQueryWithComma, getQueryWithoutAutocompletedPart} from '@
 import {buildUserReadableQueryString, getQueryWithUpdatedValues, isCannedSearchQuery, sanitizeSearchValue} from '@libs/SearchQueryUtils';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
-import type {TranslationPaths} from '@src/languages/types';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import type {SearchDataTypes} from '@src/types/onyx/SearchResults';
-import type IconAsset from '@src/types/utils/IconAsset';
-import {buildSubstitutionsMap} from './SearchRouter/buildSubstitutionsMap';
-import {getQueryWithSubstitutions} from './SearchRouter/getQueryWithSubstitutions';
-import type {SubstitutionMap} from './SearchRouter/getQueryWithSubstitutions';
-import {getUpdatedSubstitutionsMap} from './SearchRouter/getUpdatedSubstitutionsMap';
-import SearchButton from './SearchRouter/SearchButton';
-import {useSearchRouterContext} from './SearchRouter/SearchRouterContext';
-import SearchRouterInput from './SearchRouter/SearchRouterInput';
-import SearchRouterList from './SearchRouter/SearchRouterList';
-import type {SearchQueryJSON, SearchQueryString} from './types';
+import SearchTypeMenuButton from './SearchTypeMenuButton';
 
 // When counting absolute positioning, we need to account for borders
 const BORDER_WIDTH = 1;
 
 type SearchPageHeaderInputProps = {
     queryJSON: SearchQueryJSON;
+    narrowSearchRouterActive?: boolean;
+    activateNarrowSearchRouter?: () => void;
```

Comment:
possibly controversial - what do you think about naming this callback `onFocus`?